### ISSUE TITLE
Markdown link creation improvements

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/zed#5087]
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

-

### Describe the solution you'd like

1. Have link in clipboard
2. Select some text in a markdown file
3. Paste the text, Zed automatically forms a valid markdown link

---

1. Select some text in a markdown file
2. Type `[`
3. Zed automatically wraps the selected text in `[` and `]`, adds the `()` at the end, and drops the cursors in inside of it.
    - If there is a link in the clipboard, it is added in between `()`

EDIT: After typing both of these out, it seems that they are similar enough to pick just one over the other.  The first one is probably the better solution, if I had to pick one over the other.

**Comment by SamuelLHuber** [comment]
would also love if links to headlines would be suggested in editor.

like

```
# link1

[link](# <- HERE SUGGEST link1)
```

**Comment by AgapovOne** [comment]
+1 to this feature. Love it in Bear writing app. Way faster than manual work, but I can't automate it with something else macOS-wide

**Comment by marcocondrache** [comment]
@JosephTLyons I think the first solution you've described has already been implemented:

https://github.com/zed-industries/zed/blob/963204c99ddb08bd1dacbb5168194b0a39fa5ad2/crates/editor/src/editor.rs#L21897

Should this issue be closed?

**Comment by JosephTLyons** [comment]
> [@JosephTLyons](https://github.com/JosephTLyons) I think the first solution you've described has already been implemented:
> 
> [zed/crates/editor/src/editor.rs](https://github.com/zed-industries/zed/blob/963204c99ddb08bd1dacbb5168194b0a39fa5ad2/crates/editor/src/editor.rs#L21897)
> 
> Line 21897 in [963204c](/zed-industries/zed/commit/963204c99ddb08bd1dacbb5168194b0a39fa5ad2)
> 
>  fn edit_for_markdown_paste<'a>( 
> Should this issue be closed?

Yes! Thank you @marcocondrache! That PR was actually done by Cole and me! 😅
